packages/hardhat/contracts/NFTAuctionMachine.sol

```diff
@@ -178,21 +185,30 @@ contract NFTAuctionMachine is ERC721, Ownable, ReentrancyGuard, JBETHERC20Projec
         virtual
         override
         returns (string memory)
-    {   if (tokenId > totalSupply) {
-        revert INVALID_TOKEN_ID();
+    {
+        if (tokenId > totalSupply) {
+            revert INVALID_TOKEN_ID();
+        }
+
+        return metadata.tokenURI(tokenId);
+        // string memory base = _baseURI();
+        // return string(abi.encodePacked(base, "/", tokenId));
     }
-        string memory base = _baseURI();
 
-        return string(abi.encodePacked(base, "/", tokenId));
+    /**
+    @dev Updates the metadata contract address.
+    @param _metadata Address of a contract that returns tokenURI
+    */
+    function setMetadata(IMetadata _metadata) external onlyOwner {
```

Comment:
@nnnnicholas IMO we can get the `metadataFrozen` from the metadata contract right we don't need to duplicate wdyt ?

Reply:
see other comment